astropy/io/fits/connect.py

```
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
+
+    ext_data, ext_meta, ext_mask, ext_uncert : str or int, optional
+        Extensions from which to read ``data``, ``meta``, ``mask`` and
+        ``uncertainty``.
+        Default is ``0`` (data), ``0`` (meta), ``'mask'`` (mask) and
+        ``'uncert'`` (uncertainty).
+
+    kw_unit : str or None, optional
+        The header keyword which translates to the unit for the data. Set it
+        to ``None`` if parsing the unit results in a ValueError during reading.
+        Default is ``'bunit'``.
+
+    copy : bool, optional
+        Copy the data while creating the `~astropy.nddata.NDData` instance?
+        Default is ``False``.
+
+    dtype : `numpy.dtype`-like or None, optional
+        If not ``None`` the data array is converted to this dtype before
+        returning. See `numpy.ndarray.astype` for more details.
+        Default is ``None``.
+
+    kwargs_for_open :
+        Additional keyword arguments that are passed to
+        :func:`~astropy.io.fits.open` (not all of them might be possible).
+    """
+
+    # Hardcoded values to get additional information about mask and uncertainty
+    kw_hdr_masktype = 'boolean mask'
+    kw_hdr_uncerttype = {
+        'standard deviation uncertainty': StdDevUncertainty,
+        'unknown uncertainty type': UnknownUncertainty}
+
+    with fits_open(filename, mode='readonly', **kwargs_for_open) as hdus:
+        # Read the data and meta from the specified extensions
+        data = hdus[ext_data].data
+        if dtype is not None:
+            data = data.astype(dtype)
+        meta = hdus[ext_meta].header
+
+        # Read the mask and uncertainty from the specified extensions but
+        # silently fail if the extension does not exist.
+        mask = None
+        if ext_mask in hdus:
+            mask = hdus[ext_mask].data
+            # Convert it to boolean array?
+            if kw_hdr_masktype in hdus[ext_mask].header.get('comment', []):
```

This seems like a fragile way to determine the type....at the moment `NDData` assumes the mask is boolean, so why not make it `bool` by default? Or maybe add a keyword argument to specify the mask dtype if it is to be something besides boolean?